src/traits.rs

```diff
@@ -15,6 +18,13 @@ pub trait Reset {
 /// [MACD](indicators/struct.MovingAverageConvergenceDivergence.html) it is `(f64, f64, f64)` since
 /// MACD returns 3 values.
 ///
+#[cfg(feature = "serde_support")]
+pub trait Next<T>: Serialize + DeserializeOwned {
```

Comment:
Please revert changes in `Next` trait. Sorry for  not mentioning this earlier.
The trait should not depend on other traits. If caller/client needs an indicator to be serializable they have to declare requirement for `Serialize + Deserialize` traits explicitly.

As you noticed the traits in this library (Next, Reset) are pretty much thin. It's done so by design with respect to [Interface segregation principle](https://en.wikipedia.org/wiki/Interface_segregation_principle)

It could be that someone wants to implement own indicator and `Next` trait for it, but we should not force them to implement serialization if they don't need it.